### PR TITLE
DOC:matrix: Streamlined "every XX row/col" example

### PR DESCRIPTION
**Pull request by ilayn**
Continuation of #871 

A bit better example via `./list` handlers and better colorization thanks to @Mo-Gul's suggestion.

### CODE REVIEW

File: doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex
Comment (Mo-Gul) [inline doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex:521]: Again I suggest to change some colors. Here a full replacement of the `codeexample`, because I want/need to define a color upfront, so I encapsulated it in braces.

```latex
{
\definecolor{ocher}{RGB}{230,230,150}
\begin{codeexample}[
    preamble={\usetikzlibrary{matrix,fit}},
    pre={\definecolor{ocher}{RGB}{230,230,150}},
]
\begin{tikzpicture}[
  font=\sffamily,
  striped col/.style={column #1/.append style={
        every even row/.style={nodes={fill=ocher}}}},
  head color/.style args={#1/#2}{column #1/.append style={
        row 1/.append style={nodes={fill=#2}}}}
]

\matrix [
   matrix of nodes, nodes in empty cells,
   nodes={text width=2cm, align=center,
          minimum height=1.5em, anchor=center},
   striped col/.list={1,...,5}, % add striped col style to all cols
   column 1/.style={ % Override stripes and modify the feature column
     row 1 column 1/.style={nodes={fill=none, draw=none}},
     nodes={fill=ocher!80!black, inner ysep=0},
   },
   % modify headers first via common styles and then specific colors
   row 1/.style={nodes={text depth=0.2ex, text width=2cm, text=white}},
   head color/.list={2/orange,3/teal,4/cyan,5/magenta}
  ] (m)
  {
            & Basic     & Standard   & Professional & Enterprise \\
  Feature A & $\bullet$ & $\bullet$  & $\bullet$    & $\bullet$  \\
  Feature B & $\bullet$ & $\bullet$  & $\bullet$    & $\bullet$  \\
  Feature C &           &            &              & $\bullet$  \\
  Feature D &           & $\bullet$  & $\bullet$    & $\bullet$  \\
  Feature E &           &            & $\bullet$    & $\bullet$  \\
  };
% Add emphasis on selection by the use of "fit" library
\node[fit={(m-1-4.north west) (m-6-4.south east)},
      ultra thick, inner sep=0, rounded corners=1mm,
      draw=cyan, label={[cyan,align=center]270:Popular\\Choice!}]{};
\end{tikzpicture}
\end{codeexample}
}
```

Comment (ilayn) [reply doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex:521]: Ah for that you can also multiply the color values with 0.8 to mix with black. I'll add in a bit.

File: doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex
Comment (hmenke) [inline doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex]: It this really correct?  `256:red` has a colon but `184;green` and `184;blue` have a semicolon.

Comment (ilayn) [reply doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex]: Yes it's a rather obscure `xcolor` syntax that for some reason people don't use much. That package has a very powerful syntax parser which I think Till got many ideas from.

**Review by hmenke — DISMISSED** [review]

File: doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex
Comment (Mo-Gul) [reply doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex:521]: I still prefer my suggested color choices. Here an image showing your updated version (left) against mine  (right) as they are shown in the manual.

image showing ilayn's updated version against Mo-Gul's

- As you can/might see, your colors of column 1 and "Standard" are very similar. (That is why I replaced it with the one from "Enterprise" and have chosen `magenta` there.)
- Also the bright lime color doesn't fit in very well (especially at the default `codeexample` background color) an really "burns" in ones eyes.

Of course it is **your** example and it is your final decision which version to use.

Comment (ilayn) [reply doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex:521]: No problem. Could you please review line by line in the github review the changes? I thought I changed to what you suggested already. But apparently missed other details

File: doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex
Comment (Mo-Gul) [inline doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex:486]:
```suggestion
{
    \definecolor{ocher}{RGB}{230,230,150}
\begin{codeexample}[preamble={\usetikzlibrary{matrix,fit}}]
```

File: doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex
Comment (Mo-Gul) [inline doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex]:
```suggestion
     nodes={fill=olive, inner ysep=0},
```

File: doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex
Comment (Mo-Gul) [inline doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex]:
```suggestion
   head color/.list={2/orange,3/teal,4/cyan,5/magenta}
```

File: doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex
Comment (Mo-Gul) [inline doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex]: As you wished, here the line-by-line changes to your `codeexample`.

```suggestion
        every even row/.style={nodes={fill=olive!50}}}},
```

File: doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex
Comment (Mo-Gul) [inline doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex:521]:
```suggestion
\end{codeexample}
}
```
Hopefully now *I* didn't miss anything :)

File: doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex
Comment (ilayn) [reply doc/generic/pgf/text-en/pgfmanual-en-tikz-matrices.tex:486]: can we find a mix of known colors so that we skip this definition? It is a bit crowded already that's why I'm trying to run away from definitions

**Comment by Mo-Gul** [comment]
Ups, it seems that I have deleted your comment @ilayn. That was not intended but seemed to happen when I adjusted my suggestions, sorry.

---

Ok, I adapted my suggestions to use `olive` for the first column and `olive!50` for the `striped col` style. The picture then looks like

image showing the result after applying my suggestions

Fine with that?

**Comment by hmenke** [comment]
I should have merged this before #872, now there are merge conflicts.  Sorry about that.  @ilayn Let me know when this is ready to merge, then I'll resolve the conflicts.

**Comment by ilayn** [comment]
No problem. I rebased onto master and resolved conflicts. If @Mo-Gul is happy with the colors it should be good to go.

Sorry for the delay. Work is a bit busy these days.

**Review by Mo-Gul — APPROVED** [review]
When I am correct you have adapted my suggested colors. So of course I am happy :)

**Review by hmenke — APPROVED** [review]

**Comment by hmenke** [comment]
@ilayn I am very impressed with the exceptionally high quality of your contributions.  Therefore I would like to grant you write access to all repos, given that you are fine with that.

**Comment by ilayn** [comment]
Oh my, that's very flattering Henri, much appreciated. I have to ask though, I am really not active in TeX stuff so I'm usually passing by. I feel good when I am useful but are you OK with me being absent most of the time? I mean it is a responsibility that I should honor somehow but daily work + SciPy maintenance is keeping me quite busy. 

If you are OK with it sure I'd be grateful but please feel free to decline if that's not what you have in mind. Disagreement is always better than disappointment 😃

**Pull request merged**
